platform/openide.util/test/unit/src/org/openide/util/BaseUtilitiesTest.java

```diff
@@ -111,36 +112,104 @@ public void testIsJavaIdentifier() throws Exception {
+        assertEquals(new File("/some/path #1"), toFile(new URI("file:/some/path%20%231")));
+
+        //windows
+        assertEquals(new File("C:\\some\\path #1"), toFile(new URI("file:/C:/some/path%20%231")));
+        assertEquals(new File("C:\\some\\path"), toFile(new URI("file:/C:/some/path")));
+        assertEquals(new File("C:\\some\\path"), toFile(new URI("file:///C:/some/path")));
+        assertEquals(new File("C:\\some\\path"), toFile(new URI("file:/C:/some/path/")));
+        assertEquals(new File("\\\\server\\share\\path"), toFile(new URI("file://server/share/path")));
+        assertEquals(new File("\\\\server\\share\\path"), toFile(new URI("file:////server/share/path")));
+        assertEquals(new File("\\\\server\\share\\path #1"), toFile(new URI("file:////server/share/path%20%231")));
+    }
+
+    //--------------------------------------------------------------------------
+    public void test_toURI_throwsNullPonter_whenArgumentIsNull()
+            throws Exception {
+
+        try {
+            toURI(null);
+            fail();
+        } catch (final NullPointerException e) {
+            //good
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    public void test_toURI_returnsURI_whenGivenCorrectPath()
+          throws Exception {
+
+        if (isUnix()) {
+            assertEquals(new URI("file:/some/path"), toURI(new File("/some/path")));
+            assertEquals(new URI("file:///some/path"), toURI(new File("/some/path")));
+            assertEquals(new URI("file:/some/path"), toURI(new File("/some/path/")));
+            assertEquals(new URI("file:/some/path%20%231"), toURI(new File("/some/path #1")));
+        }
+        if (isWindows()) {
+            assertEquals(new URI("file:/C:/some/path%20%231"), toURI(new File("C:\\some\\path #1")));
+            assertEquals(new URI("file:/C:/some/path"), toURI(new File("C:\\some\\path")));
+            assertEquals(new URI("file:///C:/some/path"), toURI(new File("C:\\some\\path")));
+            assertEquals(new URI("file:/C:/some/path"), toURI(new File("C:\\some\\path\\")));
+            assertEquals(new URI("file://server/share/path"), toURI(new File("\\\\server\\share\\path")));
+            assertEquals(new URI("file://server/share/path"), toURI(new File("\\\\server\\share\\path\\")));
+            assertEquals(new URI("file://server/share/path%20%231"), toURI(new File("\\\\server\\share\\path #1")));
         }
-        String s = BaseUtilities.toURI(getWorkDir()).toString();
-        assertTrue(s, s.endsWith("/"));
-        URI jar = BaseUtilities.toURI(new File(getWorkDir(), "some.jar"));
-        URI jarN = jar.resolve("some.jar");
-        assertEquals(jar, jarN);
-        URI jarR = new URI("jar:" + jar + "!/");
-        URI jarNR = new URI("jar:" + jarN + "!/");
-        assertEquals("#214131: equal even when wrapped", jarR, jarNR);
```

Comment:
this is a regression test for https://bz.apache.org/netbeans/show_bug.cgi?id=214131

please don't remove it.

Reply:
I created seperate test for it 
test_toURI_producesURI_wchichResolvesAndWrapsProperly

Reply:
awesome